sdk/inc/azure/core/az_json.h

```diff
@@ -227,8 +238,37 @@ AZ_NODISCARD AZ_INLINE az_result az_json_builder_init(
   return AZ_OK;
 }
 
+// TODO: Should this accept void* user_context instead of az_allocator_context* and create the
+// allocator context internally?
 /**
- * @brief Returns the #az_span containing the JSON text written to the underlying buffer so far.
+ * @brief Initializes an #az_json_builder which writes JSON text into a destination that be
+ * non-contiguous buffers.
+ *
+ * @param[out] json_builder A pointer to an #az_json_builder instance to initialize.
+ * @param[in] first_destination_buffer An #az_span over the byte buffer where the JSON text is to be
+ * written at the start.
+ * @param[in] allocator_callback An #az_span_allocator_fn callback function that provides the
+ * destination span to write the JSON text to once the previous buffer is full or too small to
+ * contain the next token.
+ * @param[in] context An #az_allocator_context containing user-defined and other fields that are
+ * passed through to the #az_span_allocator_fn.
+ * @param[in] options __[nullable]__ A reference to an #az_json_builder_options
+ * structure which defines custom behavior of the #az_json_builder. If `NULL` is passed, the builder
+ * will use the default options (i.e. #az_json_builder_options_default()).
+ *
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if the az_json_builder is initialized successfully
+ */
+AZ_NODISCARD az_result az_json_builder_init_chunked(
+    az_json_builder* json_builder,
+    az_span first_destination_buffer,
+    az_span_allocator_fn allocator_callback,
+    az_allocator_context* context,
```

Comment:
This isn't right. The customer's context is a void*.
But, when you invoke their callback, you pass them an allocator_context which has 1 field which is the customer's void* context.

So, only the callback ever sees an allocator_context.